janusgraph-core/src/main/java/org/janusgraph/diskstorage/Mutation.java

```diff
@@ -35,10 +36,27 @@
     public Mutation(List<E> additions, List<K> deletions) {
         Preconditions.checkNotNull(additions);
         Preconditions.checkNotNull(deletions);
-        if (additions.isEmpty()) this.additions=null;
-        else this.additions = new ArrayList<>(additions);
-        if (deletions.isEmpty()) this.deletions=null;
-        else this.deletions = new ArrayList<>(deletions);
+        if (additions.isEmpty()){
+            this.additions=null;
+        } else {
+            this.additions = new ArrayList<>(additions);
+        }
+        if (deletions.isEmpty()){
+            this.deletions=null;
+        } else {
+            this.deletions = new ArrayList<>(deletions);
+        }
+    }
+
+    public Mutation(CopyArrayListFunction<E> additionsCopyFunction, CopyArrayListFunction<K> deletionsCopyFunction) {
```

Comment:
Why not use a `Supplier<List<E>>` instead of `CopyArrayListFunction<E>`?

Reply:
This is very good suggestion. Changed to Supplier